docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,259 @@
+
+When designing a custom board based on an existing Mbed Enabled board, you may need to make configuration and software changes to adapt Mbed OS to run on your new board. This is called a software port. The necessary changes may include adaptations for the unique design choices you have made for your new board, such as clocking, pin connections and peripheral usage. This can often be accomplished by adding configuration and source files to an Mbed OS-based application project without the need to modify files within `mbed-os`, itself. If you don't plan to push your changes to the upstream `mbed-os` repository, this simplifies your software configuration management by allowing you to keep your new files separate.
+
+Mbed OS supports target inheritance, which allows you to extend an existing microcontroller (MCU) target and just add software and configurations required for your board. You have the ability to add a file named `custom_target.json` to your project, which can store your custom target configurations without the need to modify `targets.json`.
+
+This tutorial provides a quick overview on how to create a custom port using `custom_target.json`. It covers the most common methods. For detailed information covering all the ways you can configure targets, go to [adding and configuring targets](../reference/adding-and-configuring-targets.html).
+
+## Extending an existing MCU target configuration
+
+As an example, consider a situation in which you are creating a new board based on an existing Mbed Enabled board. For this tutorial, consider a new board called `ImaginaryBoard` that is based on [DISCO-L475VG-IOT01A](https://os.mbed.com/platforms/ST-Discovery-L475E-IOT01A/).
+
+`ImaginaryBoard` shares most of the same features of DISCO-L475VG-IOT01A, but it does not use `AnalogOut`, `AnalogIn`, `CAN` or `USB`. Some pins are also connected differently on the new board.
+
+Assuming you have Mbed CLI installed, follow these steps to make the port.
+
+1. (Optional) Create a new Mbed program (for example, `mbed-os-imaginary-port`).
+
+   If you don't already have an Mbed program on your computer, at a command terminal, run this Mbed CLI command:
+
+   ```
+   mbed new --program mbed-os-imaginary-port
+   ```
+  
+   This command imports `mbed-os` from the [official Mbed OS source repository](https://github.com/armmbed/mbed-os) into a new directory called `mbed-os-imaginary-port`.
+
+   Now, change directories into your new project:  
+
+  ```
+  cd mbed-os-imaginary-port
+  ```
+
+1. Create a new file named `custom_targets.json` at the same level as the `mbed-os` directory:
+
+   Inspect the contents of `mbed-os/targets/targets.json`. For this example, search for `DISCO_L475VG_IOT01A`.
+
+   Copy the contents from the `DISCO_L475VG_IOT01A` section into your `custom_targets.json` file. Be sure to include brackets `{ }` surrounding the content.
+
+   Then make changes for your board. For example, after making changes, the full contents look like this:
```

Comment:
Actually, you can consider having all the previous steps as a sort of "getting ready" bit, and then the interesting bits that follow them (where you actually change the target file) are just a whole new section, with a nice header indicating that